zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/handlers/FlowNodeInstanceFromIncidentHandlerTest.java

```diff
@@ -42,22 +45,45 @@ public void testGetEntityType() {
   }
 
   @Test
-  public void testHandlesRecord() {
-    final Record<IncidentRecordValue> incidentRecord = factory.generateRecord(ValueType.INCIDENT);
-    assertThat(underTest.handlesRecord(incidentRecord)).isTrue();
+  public void shouldHandleRecord() {
+    final Set<IncidentIntent> intents2Handle =
+        Set.of(IncidentIntent.CREATED, IncidentIntent.MIGRATED);
+
+    intents2Handle.stream()
+        .forEach(
+            intent -> {
+              final Record<IncidentRecordValue> incidentRecord =
+                  factory.generateRecord(ValueType.INCIDENT, r -> r.withIntent(intent));
+              assertThat(underTest.handlesRecord(incidentRecord))
+                  .as("Handles intent %s", intent)
+                  .isTrue();
+            });
+  }
+
+  @Test
+  public void shouldNotHandleRecord() {
+    final Record<IncidentRecordValue> incidentRecord =
+        factory.generateRecord(ValueType.INCIDENT, r -> r.withIntent(IncidentIntent.RESOLVED));
+    assertThat(underTest.handlesRecord(incidentRecord)).isFalse();
```

Comment:
🔧 I know we have this in all the handler tests, but I would prefer that we stick to our given/when/then sectioning. This allows much easier to skim as reader what is actually tested.